File: playlist-service/internal/domain/entity/playlist.go
package entity

import "time"

type Playlist struct {
	ID          string
	UserID      string
	Name        string
	Description string
	SongCount   int
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

type PlaylistSong struct {
	PlaylistID string
	SongID     string
	Position   int
	AddedAt    time.Time
}

File: frontend/src/types/playlist.ts
import type { Track } from './music'

export interface Playlist {
  id: string
  name: string
  description: string
  coverUrl: string
  trackIds: string[]
  tracks?: Track[]
  ownerId: string
  ownerName?: string
  isPublic: boolean
  followerCount: number
  createdAt: string
  updatedAt: string
}

export interface PlaylistCreateInput {
  name: string
  description?: string
  coverUrl?: string
  isPublic?: boolean
}

export interface PlaylistUpdateInput {
  name?: string
  description?: string
  coverUrl?: string
  isPublic?: boolean
}

export interface PlaylistAddTrackInput {
  playlistId: string
  trackId: string
}
